src/java.base/share/classes/java/lang/StringUTF16.java

```diff
@@ -59,9 +54,9 @@ public static int newBytesLength(int len) {
         if (len < 0) {
             throw new NegativeArraySizeException();
         }
-        if (len > MAX_LENGTH) {
+        if (len >= MAX_LENGTH) {
             throw new OutOfMemoryError("UTF16 String size is " + len +
-                                       ", should be less than " + MAX_LENGTH);
+                                       ", should be less than or equal to " + MAX_LENGTH);
```

Comment:
```suggestion
                                       ", should be less than " + MAX_LENGTH);
```

Reply:
I second @ExE-Boss's comment.

Reply:
Please re-check the logic.
```
byte[] x = new byte[Integer.MAX_INTEGER];  
// throws java.lang.OutOfMemoryError: Requested array size exceeds VM limit
```
If `len  == StringUTF16.MAX_LENGTH`, then `len << 1` would be == Integer.MAX_VALUE.
And new byte[len << 1] would throw.

Note the change in the comparison from ">" to ">=".
The error message is describing the allowed range.